src/test/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/RouteCostCalculatorTest.java

```diff
@@ -0,0 +1,107 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.opentripplanner.transit.model._data.TransitModelForTest.id;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.mappers.McCostParamsMapper;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.test.support.VariableSource;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.raptor._data.transit.TestTripSchedule;
+import org.opentripplanner.transit.raptor.api.transit.CostCalculator;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransferConstraint;
+
+public class RouteCostCalculatorTest {
+
+  private static final int BOARD_COST_SEC = 5;
+  private static final int TRANSFER_COST_SEC = 2;
+  private static final double WAIT_RELUCTANCE_FACTOR = 0.5;
+  static final int UNPREFERRED_ROUTE_PENALTY = 300;
+  private static final FeedScopedId UNPREFERRED_ROUTE_ID = id("999");
+  private static final FeedScopedId DEFAULT_ROUTE_ID = id("101");
+
+  private final DefaultCostCalculator<TestTripSchedule> defaultCostCalculator = new DefaultCostCalculator(
+    BOARD_COST_SEC,
+    TRANSFER_COST_SEC,
+    WAIT_RELUCTANCE_FACTOR,
+    null,
+    null
+  );
+
+  private final RouteCostCalculator<TestTripSchedule> routeCostCalculator = createRouteCostCalculator();
+  private final TestTripSchedule.Builder scheduleBuilder = TestTripSchedule.schedule("12:00 12:01");
+
+  static Stream<Arguments> testCases = Stream.of(
+    Arguments.of(DEFAULT_ROUTE_ID, 0),
+    Arguments.of(UNPREFERRED_ROUTE_ID, UNPREFERRED_ROUTE_PENALTY)
+  );
+
+  @ParameterizedTest(name = "traversing route '{0}' should add an extra cost of {1}")
```

Comment:
Does using the single quotes (') work for you? I couldn't actually make it work and had to remove the single quotes.